.github/workflows/test.yml

```diff
@@ -0,0 +1,72 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# A workflow that runs tests on every new pull request
+name: Run unit tests
+
+on:
+  repository_dispatch:
+    types: [test]
+  push:
+    branches-ignore: ['gh-pages']
+  pull_request:
+    branches-ignore: ['gh-pages']
+  workflow_dispatch:
+  workflow_call:
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    steps:
+    # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+    - name: Checkout Repo
+      uses: actions/checkout@v4
+
+    - name: Gradle Wrapper Validation
+      uses: gradle/actions/wrapper-validation@v3
+
+    - name: Set up JDK 17
```

Comment:
This seems weird to me.  This says JDK 17, but we seem to be setting version 21 below